File: tusow/src/main/java/it/unibo/coordination/tusow/linda/TupleSpaces.java
package it.unibo.coordination.tusow.linda;

import it.unibo.coordination.linda.logic.LogicSpace;
import it.unibo.coordination.linda.text.TextualSpace;

import java.util.HashMap;
import java.util.Map;

public final class TupleSpaces {

    private TupleSpaces() {}

    private static Map<String, LogicSpace> logicSpaces = new HashMap<>();
    private static Map<String, TextualSpace> textualSpaces = new HashMap<>();

    public static LogicSpace getLogicSpace(String name) {
        if (!logicSpaces.containsKey(name)) {
            logicSpaces.put(name, LogicSpace.local(name));
        }
        return logicSpaces.get(name);
    }

    public static TextualSpace getTextualSpace(String name) {
        if (!textualSpaces.containsKey(name)) {
            textualSpaces.put(name, TextualSpace.local(name));
        }
        return textualSpaces.get(name);
    }
}

File: tusow/src/main/kotlin/it/unibo/coordination/tusow/linda/TupleSpaces.kt
package it.unibo.coordination.tusow.linda

import it.unibo.coordination.linda.logic.LogicSpace
import it.unibo.coordination.linda.text.TextualSpace
import java.util.*

object TupleSpaces {

    private val logicSpaces: MutableMap<String, LogicSpace> = HashMap()

    private val textualSpaces: MutableMap<String, TextualSpace> = HashMap()

    @JvmStatic
    fun getLogicSpace(name: String): LogicSpace {
        if (!logicSpaces.containsKey(name)) {
            logicSpaces[name] = LogicSpace.local(name)
        }
        return logicSpaces[name]!!
    }

    @JvmStatic
    fun getTextualSpace(name: String): TextualSpace {
        if (!textualSpaces.containsKey(name)) {
            textualSpaces[name] = TextualSpace.local(name)
        }
        return textualSpaces[name]!!
    }
}